libvast/builtins/query-languages/vastql.cpp

```diff
@@ -0,0 +1,36 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2022 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <vast/concept/parseable/vast/expression.hpp>
+#include <vast/error.hpp>
+#include <vast/plugin.hpp>
+
+namespace vast::plugins::vastql {
+
+class plugin final : public virtual query_language_plugin {
+  caf::error initialize(data) override {
+    return caf::none;
+  }
+
+  [[nodiscard]] std::string_view name() const override {
+    return "VASTQL";
+  }
+
+  [[nodiscard]] caf::expected<expression>
+  make_query(std::string_view query) const override {
+    auto result = expression{};
+    if (parsers::expr(query, result))
+      return result;
+    return caf::make_error(ec::invalid_query,
+                           fmt::format("not a valid query: '{}'", query));
+  }
+};
+
+} // namespace vast::plugins::vastql
+
+VAST_REGISTER_PLUGIN(vast::plugins::vastql::plugin)
```

Comment:
Nice. Please also add a docs page similar in nature to the Sigma frontend. We actually link to `VASTQL` several times in the docs. It might be worthwhile git-grepping through these places and decide whether you want to link to the plugin docs or the main language docs.

NB: Should QL really be PL soon? I could imagine that our pipeline language will be more than just queries, as it will also connect security tools.

Reply:
I went through all the other things, but this is definitely not something I'm doing on the side. I'd prefer to do this in a follow-up PR. If you agree, then please write a story for it and resolve this; other than that the PR is now on auto-merge.